ndb/src/google/cloud/ndb/model.py

```diff
@@ -1407,10 +1409,212 @@ def __init__(self, *args, **kwargs):
         raise NotImplementedError
 
 
+class _CompressedValue:
+    """A marker object wrapping compressed values.
+
+    Args:
+        z_val (bytes): A return value of ``zlib.compress``.
+    """
+
+    __slots__ = ("z_val",)
+
+    def __init__(self, z_val):
+        if not isinstance(z_val, bytes):
+            raise TypeError("Only bytes accepted", z_val)
+
+        self.z_val = z_val
+
+    def __repr__(self):
+        return "_CompressedValue({!r})".format(self.z_val)
+
+    def __eq__(self, other):
+        """Compare two compressed values."""
+        if not isinstance(other, _CompressedValue):
+            return NotImplemented
+
+        return self.z_val == other.z_val
+
+    def __ne__(self, other):
+        """Inequality comparison operation."""
+        return not self == other
+
+    def __hash__(self):
+        raise TypeError("_CompressedValue is not immutable")
+
+
 class BlobProperty(Property):
-    __slots__ = ()
+    """A property that contains values that are byte strings.
 
-    def __init__(self, *args, **kwargs):
+    .. note::
+
+        Unlike most property types, a :class:`BlobProperty` is **not**
+        indexed by default.
+
+    Args:
+        name (str): The name of the property.
+        compressed (bool): Indicates if the value should be compressed (via
+            ``zlib``).
+        indexed (bool): Indicates if the value should be indexed.
+        repeated (bool): Indicates if this property is repeated, i.e. contains
+            multiple values.
+        required (bool): Indicates if this property is required on the given
+            model type.
+        default (bytes): The default value for this property.
+        choices (Iterable[bytes]): A container of allowed values for this
+            property.
+        validator (Callable[[Property, Any], bool]): A validator to be used
+            to check values.
+        verbose_name (str): A longer, user-friendly name for this property.
+        write_empty_list (bool): Indicates if an empty list should be written
+            to the datastore.
+
+    Raises:
+        NotImplementedError: If the property is both compressed and indexed.
+    """
+
+    _indexed = False
+    _compressed = False
+
+    def __init__(
```

Comment:
Copying the entire argument list from `Property` is brittle / bad software design, but it's "necessary" to get `__repr__` to work correctly. I also think the trade-off is worth it since `BlobProperty` is nearly a decade old (i.e. the interface is set in stone).

Reply:
OK.